api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/DetermineEntitlementMessageProcessor.java

```diff
@@ -65,7 +69,10 @@ public MessageStatus processMessage(Message message) {
 
         paymentCycleService.updatePaymentCycle(currentPaymentCycle, decision);
 
-        if (decision.getEligibilityStatus() == ELIGIBLE) {
+        if (decision.getVoucherEntitlement().getTotalVoucherValueInPence() == 0) {
```

Comment:
This should probably be checking the total number of vouchers rather than their total worth.